onnxruntime/core/providers/nnapi/nnapi_builtin/builders/shaper.cc

```diff
@@ -388,6 +386,31 @@ Status Shaper::ConcatImpl(const std::vector<std::string>& input_names,
   return Status::OK();
 }
 
+Status Shaper::SplitImpl(const std::string& input_name, int32_t axis, int32_t count,
+                         const std::vector<std::string>& output_names) {
+  ORT_RETURN_IF_NOT(count > 0, "count [", count, "] must be greater than 0");
+  ORT_RETURN_IF_NOT(static_cast<size_t>(count) == output_names.size(),
+                    "count [", count, "] must match the number of outputs [", output_names.size(), "]");
+
+  const auto& input_shape = shape_map_.at(input_name);
```

Comment:
Are these checks covered by the ONNX type/shape inferencing or are there scenarios where that would exit prior to the validation being done?

e.g.
[https://github.com/onnx/onnx/blob/190b8802472b61303b0f5cb306685ef53eced1e7/onnx/defs/tensor/defs.cc#L632](https://github.com/onnx/onnx/blob/190b8802472b61303b0f5cb306685ef53eced1e7/onnx/defs/tensor/defs.cc#L632)

Older versions are in old.cc: [https://github.com/onnx/onnx/blob/190b8802472b61303b0f5cb306685ef53eced1e7/onnx/defs/tensor/old.cc#L525](https://github.com/onnx/onnx/blob/190b8802472b61303b0f5cb306685ef53eced1e7/onnx/defs/tensor/old.cc#L525)

Reply:
this could also be used to determine the shapes of intermediate NNAPI operands (e.g., within the batch MatMul composed operations). if we are adding a NNAPI split corresponding directly to an ONNX split, then the checks here may already have been done by ONNX.

will note that in this limited implementation it doesn't allow negative axis values, whereas the ONNX checks do. `count` is probably redundant - it should always be `output_names.size()`.

Reply:
ended up removing most of the checks